lib/services/messenger/translators/github.ts

```diff
@@ -0,0 +1,303 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import * as Promise from 'bluebird';
+import {
+	IssuesCreateCommentParams,
+	IssuesGetCommentsParams,
+} from 'github';
+import * as _ from 'lodash';
+import {
+	GithubConstructor,
+	GithubListenerConstructor,
+} from '../../github-types';
+import {
+	EmitInstructions,
+	MessengerAction,
+	MessengerBaseIds,
+	MessengerConstructor,
+	MessengerEvent,
+	SourceDescription,
+	TransmitInformation,
+	UpdateThreadResponse,
+} from '../../messenger-types';
+import {
+	ServiceScaffoldEvent,
+} from '../../service-scaffold-types';
+import {
+	ServiceEvent,
+	ServiceType,
+} from '../../service-types';
+import {
+	Translator,
+	TranslatorError,
+} from './translator';
+import {
+	MetadataEncoding,
+	TranslatorScaffold,
+} from './translator-scaffold';
+import {
+	EmitConverters,
+	MetadataConfiguration,
+	ResponseConverters,
+	TranslatorErrorCode,
+} from './translator-types';
+
+/**
+ * Class to enable the translating between messenger standard forms and service
+ * specific forms.
+ */
+export class GithubTranslator extends TranslatorScaffold implements Translator {
+	/**
+	 * Translations from GitHub target types into a singular form that can be used in a sentence.
+	 * e.g. the 'issue' in 'issue closed'.
+	 */
+	private static typeTranslations = {
+		issues: 'issue',
+		pull_request: 'pull request',
+	};
+
+	/**
+	 * Events on GitHub that may be used as a comment, rather than as their event description.
+	 */
+	private static commentTypes = ['issue opened', 'issue_comment created', 'pull request opened'];
+
+	/**
+	 * Convert a set of IDs from Messenger format to GitHub format.
+	 * @param ids  IDs in the form { flow, thread }
+	 * @returns    IDs in the form { owner, repo, number }
+	 */
+	private static splitIds(ids: MessengerBaseIds): { owner: string, repo: string, number?: number } {
+		const splitFlow = ids.flow.split('/');
+		const thread = ids.thread ? parseInt(ids.thread, 10) : undefined;
+		return {
+			owner: splitFlow[0],
+			repo: splitFlow[1],
+			number: thread,
+		};
+	}
+
+	/**
+	 * Converts a provided message object into instructions to read a thread for connections.
+	 * @param message  object to analyse.
+	 * @returns        Promise that resolves to emit instructions.
+	 */
+	private static readConnectionIntoEmit(message: TransmitInformation): Promise<EmitInstructions> {
+		const splitIds = GithubTranslator.splitIds(message.target);
+		const thread = splitIds.number;
+		if (!thread) {
+			return Promise.reject(new TranslatorError(
+				TranslatorErrorCode.IncompleteTransmitInformation, 'Cannot search for connections without a thread.'
+			));
+		}
+		const payload: IssuesGetCommentsParams = _.merge({ per_page: 100, number: thread }, splitIds);
```

Comment:
GH Service Emitter handles paging for you. I must be missing something, cause I can't see any handler for if there's more than 100 returned comments.

Reply:
Follow me on another journey...
1) `translators/github.ts:137-140` creates a path to an sdk method and a payload, which goes to `messenger.ts:89`
2) `messenger.ts:96-105` bind this to the sdk and bundle it into a context for `messenger.ts:106`
3) `messenger.ts:106` in turn calls the emitter's sendData (in this case `services/github.ts:244`, which deals with pagination.